src/db.rs

```diff
@@ -1854,12 +1858,47 @@ mod tests {
 				expected_count -= 1;
 			}
 		}
-		assert_eq!(col_stats.total_values, expected_count);
+		if ref_counted {
+			let mut state: BTreeMap<Vec<u8>, Option<(Vec<u8>, usize)>> = Default::default();
+			for (k, v) in change_set.iter() {
+				let mut remove = false;
+				let mut insert = false;
+				match state.get_mut(k) {
+					Some(Some((_, counter))) =>
+						if v.is_some() {
+							*counter += 1;
+						} else if *counter == 1 {
+							remove = true;
+						} else {
+							*counter -= 1;
+						},
+					Some(None) | None =>
+						if v.is_some() {
+							insert = true;
+						},
+				}
+				if insert {
+					state.insert(k.clone(), v.clone().map(|v| (v, 1)));
+				}
+				if remove {
+					state.remove(k);
+				}
+			}
+			for (key, value) in state.into_iter() {
```

Comment:
Nitpicking: `.into_iter()` is not needed here. It's what for loop does by default.

Reply:
Thanks, old habit I want to change, funny that's something I would expect clippy to find.